core-rust/state-manager/src/jni/state_computer.rs

```diff
@@ -263,9 +264,10 @@ impl From<JavaCommitRequest> for CommitRequest {
     }
 }
 
-#[derive(Debug, Decode, Encode, Categorize)]
+#[derive(Debug, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
```

Comment:
Out of interest, why did you have to change this to `ScryptoX`? Is there something that is only in the Scrypto model embedded in these types?

Reply:
`PreviousVertex` was using `ScryptoCategorize` and hence forced `PrepareRequest` to have the same.
It's kind of confusing what exactly each of the versions does so I just went with the flow :).

I've changed them to use `Decode`/`Encode`.